src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -81,47 +83,57 @@ public function getMarking($subject)
      */
     public function can($subject, $transitionName)
     {
-        $transitions = $this->definition->getTransitions();
-        $marking = $this->getMarking($subject);
+        return 0 === count($this->buildTransitionBlockerList($subject, $transitionName));
```

Comment:
Since you use this logic in quite a few places, it would be nice to have a TransitionBlockerList::isEmpty